pages/index.js

```diff
@@ -1,12 +1,14 @@
 import Header from "@/components/Header/Header";
 import TaskList from "@/components/TaskList/TaskList";
+import CreateTaskForm from "@/components/CreateTaskForm/CreateTaskForm";
 
-export default function HomePage({ sortedDefaultTasks }) {
+export default function HomePage({ currentTasks, onCreateTask }) {
   return (
     <>
       <Header />
       <main>
-        <TaskList sortedDefaultTasks={sortedDefaultTasks} />
+        <TaskList currentTasks={currentTasks} />
+        <CreateTaskForm onCreateTask={onCreateTask} />
```

Comment:
I suggest deciding based on the createMode whether you want to display the form or the button:

```suggestion
{createMode ? (
          <CreateTaskForm
            onCreateTask={(taskData) => {
              onCreateTask(taskData);
              setCreateMode(false);
            }}
            onCancel={handleCancel}
          />
        ) : (
          <StyledCreateButton onClick={() => setCreateMode(true)}>
            +
          </StyledCreateButton>
        )}
```

This will make your form a lot more reusable, since it no longer has a state and does not contain that button anymore.

```js
onCreateTask={(taskData) => {
              onCreateTask(taskData);
              setCreateMode(false);
            }}
```

I wrote the above code inline, but you can also consider making this a helper function for more readability.